internal/provider/resource_gitlab_topic.go

```diff
@@ -125,27 +126,43 @@ func resourceGitlabTopicUpdate(ctx context.Context, d *schema.ResourceData, meta
 }
 
 func resourceGitlabTopicDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
-
+	client := meta.(*gitlab.Client)
+	topicID, err := strconv.Atoi(d.Id())
+	if err != nil {
+		return diag.Errorf("Failed to convert topic id %s to int: %s", d.Id(), err)
+	}
 	softDestroy := d.Get("soft_destroy").(bool)
 
-	if !softDestroy {
-		return diag.Errorf("Destroying a topic is not yet supported. You can set soft_destroy=true to suppress this error")
+	deleteNotSupported, err := isGitLabVersionLessThan(client, "14.9")()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	if !softDestroy && deleteNotSupported {
+		return diag.Errorf("GitLab 14.9 introduced the proper deletion of topics. Set `soft_destroy = true` to empty out a topic instead of deleting it.")
 	}
 
-	log.Printf("[WARN] Not deleting gitlab topic %s. Instead emptying its description", d.Id())
+	// NOTE: the `soft_destroy` field is deprecated and will be removed in a future version.
+	//       It was only introduced because GitLab prio to 14.9 didn't support topic deletion.
```

Comment:
nit that doesn't need to block:
```suggestion
	//       It was only introduced because GitLab prior to 14.9 didn't support topic deletion.
```

Reply:
good catch, thanks!